python/tvm/micro/interface_api.py

```diff
@@ -59,7 +60,8 @@ def generate_c_interface_header(module_name, inputs, outputs, output_path):
         _emit_brief(header_file, module_name, "Input tensor pointers")
         header_file.write(f"struct {mangled_name}_inputs {{\n")
         for input_name in inputs:
-            header_file.write(f"  void* {input_name};\n")
+            sanitized_input_name = re.sub(r"\W+", "_", input_name)
```

Comment:
As this will generate one `_` for each sequence of one or more non-alphanumerical characters, is there any **realistic** chance this accidentally creates the same C name for two different input sequences (think `index-x`, `index--x`)?

Is this something that needs to be checked? I'm happy in case this is not something we expect to happen and leave as is.

Reply:
This is a good spot @leandron! When compared with the C++ implementation (`std::replace_if(sanitised_input.begin(), sanitised_input.end(), ::ispunct, '_');`) it will produce a different output as well I think, `x::2` would be `x__2` in C++ and `x_2` in Python?

@grant-arm can we add some test cases for multiple punctuation?

Reply:
Thanks @leandron & @Mousius. Good catch!
I've updated the python code that sanitizes the names and updated the test case to include a tensor name with multiple punctuation.

Reply:
hey guys--what if i have two input parameters e.g. `input-foo` and `input_foo`? i would like us to handle this case if possible,